provisioning/baremetal_pod.go

```diff
@@ -257,7 +268,27 @@ func newMetal3InitContainers(info *ProvisioningInfo) []corev1.Container {
 		initContainers = append(initContainers, createInitContainerStaticIpSet(info.Images, &info.ProvConfig.Spec))
 	}
 
-	initContainers = append(initContainers, createInitContainerMachineOsDownloader(info, true))
+	// If the PreProvisioningOSDownloadURLs are set, we fetch the URLs of either CoreOS ISO and IPA assets or in some
+	// cases only the IPA assets
+	liveURLs := getPreProvisioningOSDownloadURLs(&info.ProvConfig.Spec)
+	if liveURLs != nil {
+		initContainers = append(initContainers, createInitContainerMachineOsDownloader(info, strings.Join(liveURLs, ","), true, true))
+
+		// If the ISO URL is also specified, start the createInitContainerConfigureCoreOSIPA init container
+		if info.ProvConfig.Spec.PreProvisioningOSDownloadURLs.IsoURL != "" {
+			// Configure the LiveISO by embedding ignition and other startup files
+			initContainers = append(initContainers, createInitContainerConfigureCoreOSIPA(info))
```

Comment:
Add :
// If the CoreOS IPA assets are not available we will use the IPA downloader
if !isCoreOSIPAAvailable(&info.ProvConfig.Spec) {
		initContainers = append(initContainers, createInitContainerIpaDownloader(info.Images))
}
here?

Reply:
It's already there at line 288, we don't need two.